src/cryptography/x509.py

```diff
@@ -976,6 +982,11 @@ def __ne__(self, other):
         return not self == other
 
 
+class SubjectAlternativeName(GeneralNames):
```

Comment:
https://www.youtube.com/watch?v=3MNVP9-hglc outlines my feelings on the
matter.

On Wed, May 27, 2015 at 11:47 AM, Erik Trauschke <notifications@github.com>
wrote:

> In src/cryptography/x509.py
> <https://github.com/pyca/cryptography/pull/1956#discussion_r31148034>:
>
> > @@ -976,6 +982,11 @@ def __ne__(self, other):
> >          return not self == other
> >
> >
> > +class SubjectAlternativeName(GeneralNames):
>
> Ok, understood. Regarding the test cases, I'd assume we now have to run
> the whole set of tests for SAN on the GeneralNames class as well. Also,
> could you explain what the benefit of this solution is compared to
> subclassing?
> … <#14d960e898f81258_>
> On Wed, May 27, 2015 at 8:15 AM, Alex Gaynor <notifications@github.com>
> wrote: In src/cryptography/x509.py <
> https://github.com/pyca/cryptography/pull/1956#discussion_r31144337>: >
> @@ -976,6 +982,11 @@ def __ne__(self, other): > return not self == other >
> > > +class SubjectAlternativeName(GeneralNames): You can probably just
> simplify __init__ to construct the GeneralNames, no need for the isinstance
> check, GeneralNames.__init__ will have that. — Reply to this email directly
> or view it on GitHub <
> https://github.com/pyca/cryptography/pull/1956/files#r31144337>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1956/files#r31148034>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084